base/applications/cmdutils/more/more.c

```diff
@@ -134,28 +312,103 @@ PagePrompt(PCON_PAGER Pager, DWORD Done, DWORD Total)
+            Pager->PagerAction = MorePageActionDoNothing;
+            return TRUE;
+    }
+    s_chSubCommand = 0;
+
     /* Ctrl+C or Ctrl+Esc: Control Break */
-    if ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
-        ((KeyEvent.wVirtualKeyCode == L'C') &&
-         (KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED))))
+    if (fCtrl && ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
+                  (KeyEvent.wVirtualKeyCode == L'C')))
     {
         /* We break, output a newline */
         WCHAR ch = L'\n';
         ConStreamWrite(Pager->Screen->Stream, &ch, 1);
         return FALSE;
     }
 
-    /* 'Q': Quit */
-    // FIXME: Available only when command extensions are enabled.
-    if ((KeyEvent.wVirtualKeyCode == L'Q') &&
-        !(KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED)))
+    if (s_dwFlags & FLAG_E) /* If extended features are available */
     {
-        /* We break, output a newline */
-        WCHAR ch = L'\n';
-        ConStreamWrite(Pager->Screen->Stream, &ch, 1);
-        return FALSE;
+        /* 'Q': Quit */
+        if ((KeyEvent.wVirtualKeyCode == L'Q') && !fCtrl)
+        {
+            /* We break, output a newline */
+            WCHAR ch = L'\n';
+            ConStreamWrite(Pager->Screen->Stream, &ch, 1);
+            return FALSE;
+        }
+
+        /* 'F': Next file */
+        if ((KeyEvent.wVirtualKeyCode == L'F') && !fCtrl)
+        {
+            Pager->PagerAction = MorePageActionNextFile;
+            return TRUE;
+        }
+
+        /* '?': Show Options */
+        if (KeyEvent.uChar.UnicodeChar == L'?' && !fCtrl)
+        {
+            Pager->PagerAction = MorePageActionDoNothing;
+            s_nPromptID = IDS_CONTINUE_OPTIONS;
+            return TRUE;
+        }
+
+        /* [Enter] key: One line */
+        if ((KeyEvent.wVirtualKeyCode == VK_RETURN) && !fCtrl)
+        {
+            Pager->ScrollRows = 1;
+            return TRUE;
+        }
+
+        /* 'P': Display n lines */
+        if (KeyEvent.wVirtualKeyCode == L'P' && !fCtrl)
+        {
+            Pager->PagerAction = MorePageActionDoNothing;
+            s_nPromptID = IDS_CONTINUE_LINES;
+            s_chSubCommand = L'P';
+            return TRUE;
+        }
+
+        /* 'S': Skip n lines */
+        if (KeyEvent.wVirtualKeyCode == L'S' && !fCtrl)
+        {
+            Pager->PagerAction = MorePageActionDoNothing;
+            s_nPromptID = IDS_CONTINUE_LINES;
+            s_chSubCommand = L'S';
+            return TRUE;
+        }
+
+        /* '=': Show current line */
+        if (KeyEvent.uChar.UnicodeChar == L'=' && !fCtrl)
+        {
+            Pager->PagerAction = MorePageActionDoNothing;
+            s_nPromptID = IDS_CONTINUE_LINE_AT;
+            return TRUE;
+        }
+    }
+
+    /* [Space] key: One page */
+    if ((KeyEvent.wVirtualKeyCode == VK_SPACE) && !fCtrl)
+    {
+        Pager->ScrollRows = Pager->ScreenRows - 1;
+        return TRUE;
     }
```

Comment:
Please move this before the treatment of the extended features,

Reply:
OK.